tools/tpm2_ptool.py

```diff
@@ -1063,175 +1291,90 @@ def generate_options(self, group_parser):
+        # the db.
+        alg = args['algorithm']
+        if alg !='rsa':
+            sys.exit('Unknown algorithm or algorithm not supported, got "%s"' % alg)
 
-        label = args['label']
-        keylabel= args['key_label']
-
-        id = args['id']
-
-        with Db(path) as db:
-
-            token = db.gettoken(label)
-
-            token_config =  dict_from_kvp(token['config'])
-
-            pobj = db.getprimary(token['pid'])
-
-            with TemporaryDirectory() as d:
-                tpm2 = Tpm2(d, path)
-
-                # Get the primary object encrypted auth value and sokey information
-                # to decode it. Based on the incoming pin
-                is_so = args['sopin'] != None
-                if is_so:
-                    pin = args['sopin']
-                    pinpobjauthkeysalt = token['sopobjauthkeysalt']
-                    pinpobjauthkeyiters = token['sopobjauthkeyiters']
-                    pinpobjauth = token['sopobjauth']
-                else:
-                    pin = args['userpin']
-                    pinpobjauthkeysalt = token['userpobjauthkeysalt']
-                    pinpobjauthkeyiters = token['userpobjauthkeyiters']
-                    pinpobjauth = token['userpobjauth']
-
-                pinpobjauthkeysalt = binascii.unhexlify(pinpobjauthkeysalt)
-                pinpobjauthkey = hash_pass(pin.encode(), iters=pinpobjauthkeyiters, salt=pinpobjauthkeysalt)
-
-                pinpobjauth = AESCipher(pinpobjauthkey['rhash']).decrypt(pinpobjauth)
-
-                # At this point we have recovered the ACTUAL auth value for the primary object, so now we
-                # can load up the seal objects
-                sealobject = db.getsealobject(token['id'])
-
-                if is_so:
-                    sealpub = sealobject['sopub']
-                    sealpriv = sealobject['sopriv']
-                    salt = sealobject['soauthsalt']
-                    iters = sealobject['soauthiters']
-                else:
-                    sealpub = sealobject['userpub']
-                    sealpriv = sealobject['userpriv']
-                    salt = sealobject['userauthsalt']
-                    iters = sealobject['userauthiters']
+        with TemporaryDirectory() as d:
+            tpm2 = Tpm2(d, path)
+            tertiarypriv, tertiarypub, tertiarypubdata = tpm2.importkey(sobjctx, sobjauth, objauth, privkey=privkey, alg=alg)
 
-                salt = binascii.unhexlify(salt)
-                sealauth = hash_pass(pin.encode(), iters, salt)['hash']
+            return (tertiarypriv, tertiarypub, tertiarypubdata)
 
-                # Load the so sealobject using the PARENTS AUTH (primaryobject)
-                sealctx = tpm2.load(pobj['handle'], pinpobjauth, sealpriv, sealpub)
+    def __call__(self, args):
+        super(self.__class__, self).__call__(args)
+        keylabel= args['key_label']
+        print('Imported key as label: "{keylabel}"'.format(keylabel=keylabel))
```

Comment:
What if keylabel isn't specified? This needs to default to rowid.

Reply:
it is done in the new_key_save method in the super calss

Reply:
> You still have make check failures.

Now that the tpm2-tools part is upstream I hope this make check failures will be fixed.

Reply:
> What if keylabel isn't specified? This needs to default to rowid.

I see that the key-lable changes that was added to the addkey command had the label default to the 'id'. If you say it should default to rowid then how will the user know what the rowid is when they want to refer to this key?